src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -109,51 +123,59 @@ function AcceptButton ({ transaction: { id } }) {
   </Button>
 }
 
-function useOffer (id, amount, counterparty) {
-  const [offer] = useMutation(HolofuelOfferMutation)
-  return () => offer({
-    variables: { amount, counterparty, requestId: id },
-    refetchQueries: [{
-      query: HolofuelActionableTransactionsQuery
-    }]
-  })
-}
-
-function PayButton ({ transaction: { id, amount, counterparty } }) {
-  const pay = useOffer(id, amount, counterparty)
+function PayButton ({ showConfirmationModal, transaction }) {
+  // const pay = useOffer(id, amount, counterparty)
   return <Button
-    onClick={pay}
+    onClick={() => showConfirmationModal(transaction)}
     styleName='pay-button'>
     Pay
   </Button>
 }
 
-function RejectButton ({ showRejectionModal, transaction }) {
+function RejectButton ({ showConfirmationModal, transaction }) {
   return <Button
-    onClick={() => showRejectionModal(transaction)}
+    onClick={() => showConfirmationModal(transaction)}
     styleName='reject-button'>
     Reject
   </Button>
 }
 
-function ConfirmRejectionModal ({ transaction, handleClose, declineTransaction }) {
+export function ConfirmationModal ({ transaction, handleClose, declineTransaction, payTransaction }) {
   if (!transaction) return null
-
   const { id, counterparty, amount, type } = transaction
+
+  const action = type === TYPE.request ? 'pay' : 'decline'
+
+  let message, actionHook, actionParams
+  switch (action) {
+    case 'pay': {
+      actionParams = { id, amount, counterparty }
+      actionHook = payTransaction
+      message = <div styleName='modal-text' data-testid='modal-message'>Pay <span styleName='modal-counterparty'><RenderNickname agentId={counterparty} /></span><span styleName='modal-amount'>{presentHolofuelAmount(amount)} HF</span>?</div>
+      break
+    }
+    case 'decline': {
+      actionParams = id
+      actionHook = declineTransaction
+      message = <div styleName='modal-text' data-testid='modal-message'>Reject <span styleName='modal-counterparty'><RenderNickname agentId={counterparty} /></span>'s {type} of <span styleName='modal-amount'>{presentHolofuelAmount(amount)} HF</span>?</div>
+      break
+    }
+    default:
+      throw new Error('Error: Transaction action was not matched with a modal action. Current transaction action : ', action)
+  }
+
   const onYes = () => {
-    declineTransaction(id)
+    actionHook(actionParams)
     handleClose()
   }
 
   return <Modal
-    contentLabel={`Reject ${type}?`}
+    contentLabel={`${_.capitalize(action)} ${type}?`} // WAS: contentLabel={`Reject ${type}?`}
```

Comment:
I think it would be more readable just to set a `contentLabel` var in each of the two cases. So this line would just read `contentLabel={contentLabel}`, and that value is set above based on the case.